lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -647,11 +646,17 @@ Updates result of a job in the system.
 
 sub done {
     my ($self) = @_;
-
     return undef unless my $job = $self->find_job_or_render_not_found($self->stash('jobid'));
-    my $result   = $self->param('result');
-    my $reason   = $self->param('reason');
-    my $newbuild = defined $self->param('newbuild') ? 1 : undef;
+
+    my $validation = $self->validation;
+    $validation->optional('result')->in(NONE, PASSED, SOFTFAILED, FAILED, INCOMPLETE, SKIPPED);
```

Comment:
This list of results is incomplete, **e.g.** `TIMEOUT_EXCEEDED` is missing. There is the `RESULTS` constant which contains all valid results. But don't use it here. Invalid results are already sorted out in the `done` function of the job so this just duplicates the validation. Just add a comment that this is already validated in `$job->done`.

Reply:
I actually tried to use `RESULTS` at first, and it didn't work with `in`.

You have a point. It's redundant with the validation in the `Jobs` result. I was mainly adding it since it is early and it's less obvious unless you're reading the code carefully. A comment will achieve the same, though.